Bump version to 2.3.0 with python 3.14 support

AyanSinhaMahapatra opened the pull request:
This PR bumps version to 2.3.0 and:
- drops support for Python 3.9 which is end-of-life
- adds support for Python 3.14. 

- All release testing passes at https://github.com/AyanSinhaMahapatra/pyahocorasick/releases/tag/v2.3.0rc2 with required wheels.

- Also added a release script at https://github.com/WojciechMula/pyahocorasick/blob/support-py314/.github/workflows/pypi-release.yml to publish to pypi with trusted publishing. https://docs.pypi.org/trusted-publishers/

AyanSinhaMahapatra committed:
Bump version to 2.3.0 with python 3.14 support

Signed-off-by: Ayan Sinha Mahapatra <asmahapatra@aboutcode.org>

pombredanne commented on .github/workflows/pypi-release.yml:
```diff
@@ -0,0 +1,152 @@
+name: Create library release archives, create a GH release and publish PyPI wheel and sdist on tag in main branch
+
+
+# This is executed automatically on a tag in the main branch
```
Why duplicate this with the test-and-build.yml?

AyanSinhaMahapatra replied:
We still need to test the wheel builds for all os/python combinations and run the tests on each commit, this was the case before too. See https://github.com/WojciechMula/pyahocorasick/blob/v2.2.0/.github/workflows/test-and-build.yml
Previously to release a new version, we just used the wheels built from these tests, and then downloaded it, uploaded it to pypi.

For the pypi release action, this will create the wheels and sdist seperately, and then upload this to github and pypi.
This is why we need the wheels built on both actions, to test on every commit, and to build them for release.

pombredanne reviewed (changes requested):
Thanks... can you combine this in one CI job that would 1. upload artifacts 2. attempt to do a secure publoishing?

AyanSinhaMahapatra committed:
Update test scripts to remove sdist builds

We already build sdists and wheels in the pypi release script,
and in the test script we only need to build and test wheels.

Signed-off-by: Ayan Sinha Mahapatra <asmahapatra@aboutcode.org>

AyanSinhaMahapatra commented:
@pombredanne I've updated the scripts to get rid of some duplicate parts, we still need two CI jobs:
1. to test on every commit (with wheels builds to check that)
2. to release on tag push. we need to turn on trusted publishing (or use an API key) so that we don't have to release this manually.

pombredanne reviewed (approved):
Thanks. Looking great!
